File: csharp/AbstractCommands/AbstractUnaryFileCommand.cs
using System.Collections.Generic;
using Validators;

namespace CsharpCommandEngine
{
    public abstract class AbstractUnaryFileCommand: AbstractFileCommand
    {
        private const string ArgumentCountMismatchMessage = "ERROR: the command takes EXACTLY 1 argument!";

        protected new void SetupValidation()
        {
            Validation.AddValidator(new ArgumentCountValidator(Arguments, 1, ArgumentCountMismatchMessage));
        }

        public AbstractUnaryFileCommand(List<string> args) : base(args)
        {
            SetupValidation();
        }
    }
}

File: python/AbstractCommands/AbstractUnaryFileCommand.py
from .AbstractCommand import *
from Validators.ArgumentCountValidator import *
from pathlib import Path


class AbstractUnaryFileCommand(AbstractCommand):
    @staticmethod
    def _argument_count_mismatch_message():
        return "the command takes EXACTLY 1 argument!"

    def _setup_validation(self):
        self._validation.add_validator(ArgumentCountValidator(self.arguments, 1, self._argument_count_mismatch_message()))

    def __init__(self, args):
        super().__init__(args)
        self._setup_validation()
        self._file = Path(self.arguments[0]) if self.arguments.__len__() > 0 else Path("")

    @abstractmethod
    def execute(self):
        pass

